### PR TITLE
IModelJsFs check dir exists before reading

### LINKED ISSUE #4399: Reading Settings Schemas requires assets/Settings/Schemas directory
**Describe the bug**
As of iTwin 3.3.x, a directory in the backend assets named Settings/Schemas must exist to avoid a run-time exception. Since everything works when this directory is empty, it seems to me that requiring the directory to exist is a bug. I believe that the error is in the following line of code:

https://github.com/iTwin/itwinjs-core/blob/8735b84372ca7ce781f6bc5fc98b4a0d3940a782/core/backend/src/workspace/SettingsSchemas.ts#L119

Adding an exception handler around that (or a check to see if the directory exists) would presumably fix the problem.

**To Reproduce**

1. Create an app based on iTwin 3.3.x or later.
2. Make sure that the backend assets directory does not contain a Settings subdirectory.
3. Start your app.
4. Crash

**Expected behavior**
Since the schemas apparently aren't necessary in order for things to function, the directory containing the schemas should not be necessary. Right now I have to put the directory in my app's git repository and have a dummy.txt file, since git doesn't support empty directories.

**Desktop (please complete the applicable information):**

- OS: iOS
- iTwin.js Version 3.3.5

### PR DESCRIPTION
**Pull request by aruniverse**
closes #4399

### CODE REVIEW

**Review by pmconne — APPROVED** [review]
Document the change in behavior since it differs from the `fs` function.

**Comment by aruniverse** [comment]
@mergifyio backport release/3.3.x

**Comment by aruniverse** [comment]
@tcobbs-bentley please test this works with mobile before this pr completes

**Review by tcobbs-bentley — DISMISSED** [review]

File: core/backend/src/IModelJsFs.ts [hunk 1]
```diff
@@ -59,8 +59,8 @@ export class IModelJsFs {
   /** Make a copy of a file */
   public static copySync(src: string, dest: string, opts?: any): void { fs.copySync(src, dest, opts); }
 
-  /** Get the file and directory names in the specified directory. Excludes "." and "..". */
-  public static readdirSync(pathname: string): string[] { return fs.readdirSync(pathname); }
+  /** Gets the file and directory names in the specified directory. Excludes "." and "..". Returns an empty array if the specified directory does not exist. */
+  public static readdirSync(pathname: string): string[] { return fs.existsSync(pathname) ? fs.readdirSync(pathname) : []; }
```
Comment (tcobbs-bentley) [inline core/backend/src/IModelJsFs.ts:63]: I will resolve this comment after I have tested on mobile. Testing on mobile will take some time.

Comment (tcobbs-bentley) [reply core/backend/src/IModelJsFs.ts:63]: I have confirmed that this fix works on mobile in display-test-app.

**Review by tcobbs-bentley — APPROVED** [review]

**Pull request merged**